tools/loader/ImageClassifier.cpp

```diff
@@ -253,8 +285,26 @@ static void processAndPrintResultsImpl(Tensor *SMT,
   const size_t numLabels = SMT->dims()[1];
   std::vector<size_t> sliceOffset(SMT->dims().size(), 0);
 
+  // Parse provided, if any, category indices.
+  std::vector<unsigned> matchingIndices(inputImageFilenames.size(), ~0U);
```

Comment:
I think you could totally do away with this extra vector and the check for `~0U` in `matchIndices()` -- in the loop below you could just call `matchIndices()` with `expectedMatchingLabels` if `!expectedMatchingLabels.empty()`.